v3/request.go

```diff
@@ -71,28 +70,33 @@ func (l *Conn) readPacket(msgCtx *messageContext) (*ber.Packet, error) {
 	return packet, nil
 }
 
-func getReferral(err error, packet *ber.Packet) (referral string, e error) {
+func getReferral(err error, packet *ber.Packet) (referral string) {
 	if !IsErrorWithCode(err, LDAPResultReferral) {
-		return "", nil
+		return ""
 	}
 
 	if len(packet.Children) < 2 {
-		return "", fmt.Errorf("ldap: returned error indicates the packet contains a referral but it doesn't have sufficient child nodes: %w", err)
+		return ""
 	}
 
-	if packet.Children[1].Tag != ber.TagObjectDescriptor {
-		return "", fmt.Errorf("ldap: returned error indicates the packet contains a referral but the relevant child node isn't an object descriptor: %w", err)
+	children := len(packet.Children[1].Children)
+
+	if children == 0 || (packet.Children[1].TagType != ber.TypeConstructed || packet.Children[1].ClassType != ber.ClassApplication) {
+		return ""
 	}
 
 	var ok bool
 
-	for _, child := range packet.Children[1].Children {
-		if child.Tag == ber.TagBitString && len(child.Children) >= 1 {
-			if referral, ok = child.Children[0].Value.(string); ok {
-				return referral, nil
-			}
+	for i := 0; i < children; i++ {
+		if (packet.Children[1].Children[i].Tag != ber.TagBitString && packet.Children[1].Children[i].Tag != ber.TagPrintableString) ||
```

Comment:
Do you mind adding a comment or reference link to this issue, so others can understand whats going here with this giant if-exclusion. Especially regarding the edge case with OpenLDAP

Reply:
And perhaps we can use a variable here named `child := packet.Children[1].Children[i]` instead of typing the reference out every time

Reply:
Yep no worries.

Reply:
This is done. I explained both packet filtering stages with hopefully no uncertain terms.